elixir/apps/api/lib/api/client/channel.ex

```diff
@@ -836,4 +970,45 @@ defmodule API.Client.Channel do
         end
     end
   end
+
+  # Ice credentials must stay the same for all connections between client and gateway as long as they
+  # do not loose their state, so we can leverage public_key which is reset on each restart of the client
+  # or gateway.
+  defp generate_ice_credentials(client, gateway) do
+    ice_credential_seed =
+      [
+        client.id,
+        client.public_key,
+        gateway.id,
+        gateway.public_key
+      ]
```

Comment:
Should we add a domain separator here so the input to the hash function isn't just these IDs and keys?

Reply:
Do you mean add a constant? I did not add it because it doesn't give any benefits in this case.

The components are joined with `:` already.

Reply:
> Do you mean add a constant? I did not add it because it doesn't give any benefits in this case.
> 
> The components are joined with `:` already.

Not necessarily a compile-time constant but a random value per process instance for example. The credentials aren't sensitive so it might not matter, but if you prefix it with a runtime value, it becomes impossible for a client or gateway to grind for certain ICE credentials by choosing a certain public key.

Reply:
@thomaseizinger what will happen if the client reconnects to the portal (eg. while roaming) and gets new ice credentials for the new connection to the same gateway?

Reply:
With process I meant system process, i.e. when the portal as a whole starts up. It should stay constant across reconnects of the websocket.

Reply:
@thomaseizinger, these credentials aren't sensitive?